cmd/kyma/alpha/create/module/module.go

```diff
@@ -56,12 +62,13 @@ Finally, if a registry is provided, the created module is pushed.
 	cmd.Flags().BoolVarP(&o.Overwrite, "overwrite", "w", false, "overwrites the existing mod-path directory if it exists")
 	cmd.Flags().BoolVar(&o.Insecure, "insecure", false, "Use an insecure connection to access the registry.")
 	cmd.Flags().BoolVar(&o.Clean, "clean", false, "Remove the mod-path folder and all its contents at the end.")
+	cmd.Flags().BoolVar(&o.ValidateDefaultCR, "validateCR", false, "Validate the custom resource defined in the \"default.yaml\" file on demand. This validation always runs when pushing the module.")
```

Comment:
mhm, wait. Line 47 says that validation happens _before_ pushing; but this line says it is always triggered by the pushing, so it happens after pushing starts (and probably aborts the pushing if validation is unsuccessful?).

Reply:
Validation - if triggered - always executes early in the command, to "fail fast".
Validation is triggered in two cases:
 - user asked for it by setting the `validateCR` flag (on-demand validation)
 - user wants to push the artifact to the registry by configuring `--registry=https://some.registry.com` flag
 
 So if the command is configured to push the artifact - which is at the end of the command, the validation will be always triggered at the  beginning of the command (to fail fast in case of errors).
 Now we have to find a way to describe this in a nice and short way :)

Reply:
Thank you for the great explanation, this helps! 

I'll try to document that information in the `Detailed description`, see my proposal here: https://github.com/kyma-project/cli/pull/1379/files#r992524816